hyperopt/fmin.py

```diff
@@ -493,6 +494,13 @@ def fmin(
         were succesfull trails. This object shares the same structure as the space passed.
         If there were no succesfull trails, it returns None.
     """
+    if algo is None:
```

Comment:
it would have been good to write this to the docstring. Plus, now that we assume only 3.7+, we could start typehinting :)  let's think about it for next PRs